File: FingerRace/Player.m
//
//  Player.m
//  FingerRace
//
//  Created by Emmett Butler on 3/31/13.
//
//

#import "Player.h"
#import "SquareTarget.h"

@implementation Player

@synthesize color, currentTarget, touch;

+(Player *)player{
    return [[Player alloc] initWithColor:ccc3(arc4random() % 255, arc4random() % 255, arc4random() % 255)];
}

-(Player *)initWithColor:(ccColor3B)col{
    if(self=[super init]){
        self.color = col;
    }
    return self;
}

-(void)spawnNewTargetWithLayer:(CCLayer *)layer{
    CGPoint randPos = CGPointMake(arc4random() % (int)[layer boundingBox].size.width, arc4random() % (int)[layer boundingBox].size.height);
    if(self.currentTarget == nil){
        self.currentTarget = [SquareTarget spriteWithPlayer:self];
        self.currentTarget.position = randPos;
        [layer addChild:self.currentTarget];
    }
    [self.currentTarget runAction:[CCMoveTo actionWithDuration:.1 position:randPos]];
}

@end

File: FingerRace/Classes/Player.cpp
//
//  Player.cpp
//  CocosTest
//
//  Created by Evan Moore on 4/2/13.
//
//

#include "Player.h"

#import "cocos2d.h"
#import "SquareTarget.h"

Player::Player() {
    this->currentTarget = new SquareTarget();
}

void Player::initWithColor(cocos2d::ccColor3B col) {
    this->color = col;
}

void Player::spawnNewTargetWithLayer(cocos2d::CCLayer * layer) {
    this->killOldTarget();
    this->currentTarget = new SquareTarget();
    this->currentTarget->initWithColor(this->color);
    this->currentTarget->setPosition(*new cocos2d::CCPoint(arc4random() % (int)layer->boundingBox().size.width, arc4random() % (int)layer->boundingBox().size.height));
    layer->addChild(this->currentTarget);
}

void Player::killOldTarget() {
    this->currentTarget->removeFromParentAndCleanup(true);
}
